### ISSUE TITLE
[Bug]: Signature popup not showing up sometimes on iOS/Android app with WalletConnect

### ISSUE THREAD

**Issue by MananTank** [MetaMask/metamask-mobile#9116]
### Describe the bug

When connecting the MetaMask app with WalletConnect and then sending a request to sign a message, Sometimes the MetaMask app does not show the signature popup, sometimes it takes 20-30 seconds and sometimes it randomly shows the signature popup for a request that was triggered 5-10 minutes ago 

You can replicate the issue here: [Test App](https://internal-test-next-git-mnn-mm-sign-test.thirdweb-preview.com/?_vercel_share=vhxk2963vWZsEGwWLMUoUxciBobHMkCJ)

This issue is not always replicable - It only happens sometimes. If you try signing the message multiple times - you will probably be able to replicate this issue. 

I am only seeing this issue in MetaMask - The others wallets like Trust Wallet, Rainbow etc are working as expected

### Expected behavior

When sending a signature request - MetaMask app should show the signature request popup

### Screenshots/Recordings

Please see the below recording: 

You will see the MetaMask does not show the signature popup, while Trust wallet does. 

( Note: You have to try it a few times to replicate the issue - sometimes it works and sometimes it doesn't on MetaMask - while it always works for other wallets ) 

https://github.com/MetaMask/metamask-mobile/assets/22043396/389674a1-cb3e-4b11-8eac-8c6d6c2f6694

### Steps to reproduce

On your Mobile device: ( iOS preferred )

2. Go to [Test App](https://internal-test-next-git-mnn-mm-sign-test.thirdweb-preview.com/?_vercel_share=vhxk2963vWZsEGwWLMUoUxciBobHMkCJ)
3. Click on "Connect Wallet"
4. Select MetaMask and connect 
5. Go back to webpage and wait for connection to be confirmed 
6. click on "Sign Message" button -> you will be redirected to MetaMask app 
7. At this point -> MetaMask should show popup to sign the message "hello!"

If you see this popup, it means that this is working for you and the issue is not replicable. 
Try doing this a couple more times and you will be able to replicate this issue - either the Signature popup will take a really long time to show up ( 20-30 seconds ) or it won't show up at all. 

If you are still not able to replicate this issue, try connecting and then switching the network and then signing the message as mentioned in below steps: 

* Go to webpage, connect to MetaMask
* click on the button that is showing the connected wallet details 
* click on the button that is showing the connected network 
* This will open a screen to switch networks
* Click on any of the network to switch to that network
* approve the request in wallet ( We have noticed that sometimes even this does not work - MetaMask app just does not show any popup to switch network - please look into this as well )
* go back to webpage, click on "sign message"

### Version

7.18.0

### Build type

None

### Device

iPhone 15 Pro Max

### Operating system

iOS

**Comment by ntchungse** [comment]
same issues. If I use metamask ver 7.14., it works fine

**Comment by cashspicy** [comment]
Same issue. Updating the metamask app didnt help too.

**Comment by nfftdev** [comment]
Same issue on v7.18 and sometimes when it does popup, it doesn't redirect after signing.  Instead it will say "Please check the application".  But sometimes it redirects back to the browser as expected.

**Comment by christopherferreira9** [comment]
Hi all! 
We are aware of this issue and it they started showing up after bumping @walletconnect/se-sdk to version `1.5.0`. If you were to wait for a few minutes with the wallet open or switch between Dapp and Wallet a few times you'll end up getting the request.
We're currently in sync with WalletConnect to have all of these delays sorted and have it ready and working as soon as possible.

**Comment by DeynekaYuriy** [comment]
> I'm developing a react native application, which uses wallet connect. I face a similar issue as well. The metamask popup appears when in USB debugging mode on a physical device (note that it still fails in an emulator), but it always fails in a production build.
> 
> If you try the Rabby wallet mobile app on Android, you can see the same issue there, when you try to request for a signature when transacting.
> 
> **Versions used**
> 
> ```
> "react": "18.2.0",
> "react-native": "0.73.5",
> "viem": "1.21.4",
> "wagmi": "1.4.13"
> ```
> 
> Metamask android version: 7.18.0

Hi! 
Have you found a solution?÷

**Comment by hightouch67** [comment]
Same issue here. It randomly works. Hope this become a priority. If anyone has a workaround please share it.

**Comment by christopherferreira9** [comment]
Hi all! We've recently merged the fixes for this issue that  improves the behaviour and responsiveness when using WalletConnect on https://github.com/MetaMask/metamask-mobile/pull/9188 and https://github.com/MetaMask/metamask-mobile/pull/9150. These will be available in version `7.21.0`.
I'm closing this issue but please feel free to open a new issue whenever you see fit.

**Comment by jiaji-wei** [comment]
Hi everyone ! I still can't complete the signing. Every time I try to connect my wallet to sign, a "Return to app" message pops up, but the signing window doesn't appear. Any ideas?

**Comment by Suhel-Kap** [comment]
> Yes. Even I can't get the signing message. It has something to do with it's internal app data, if I run it fresh after clearing the app data it works but then it stops there after. Using version 7.20.1 (1314) Android platform.

These will be available in version 7.21.0 as mentioned by [christopherferreira9](https://github.com/christopherferreira9)

We'll have to wait for the update

**Comment by orhunerd** [comment]
> Hi all! We've recently merged the fixes for this issue that improves the behaviour and responsiveness when using WalletConnect on #9188 and #9150. These will be available in version `7.21.0`. I'm closing this issue but please feel free to open a new issue whenever you see fit.

Unfortunately after the 7.21.0 release the connect and sign modals never appears on my end. I started to see the "Connecting to Metamask" modal but nothing happens after that. Before this release, if you go back and forth between the app and metamask, you had a chance to see the connect or sign modal. Thanks for keeping an eye on this though.
![IMG_1694](https://github.com/MetaMask/metamask-mobile/assets/85942921/aa2d24d7-25c3-4db7-b743-2ba7dcc6165e)

**Comment by christopherferreira9** [comment]
Hi @orhunerd , are you able to share a dapp we can reproduce this with?

**Comment by orhunerd** [comment]
> Hi @orhunerd , are you able to share a dapp we can reproduce this with?

You can reproduce the issue in our app cookievault.com. After you navigate there hit connect wallet. You will see a "wallet connect" pop up. Hit metamask..

Another site you can reproduce this is tofunft.com if you hit the profile symbol on top right side of the screen, this will trigger a modal to connect a wallet. You can either choose wallet connect or metamask directly. It all results the same. You end up waiting on "Connecting to Metamask" modal on Metamask app.
I have rebooted my iPhone and repeated again it all end up the same.

**Comment by orhunerd** [comment]
@christopherferreira9 this is a screen recording about the issue
https://github.com/MetaMask/metamask-mobile/assets/85942921/e5fc4669-ab9b-47de-be2d-4d2a3f4dba1f

**Comment by orhunerd** [comment]
@christopherferreira9 our app is on react, web3modal, wallet connect, wagmi. Any suggestion for us to make this wallet connection to work better?

**Comment by VijiHithub** [comment]
**Greetings Team,**

I'm currently utilizing the Metamask app version 7.22.0 on my Android device. Unfortunately, I'm encountering an issue where the pop-ups for connect, approve, or sign transactions are not appearing. I'm working on integrating wallet connect functionalities with ethers.js. I would greatly appreciate it if you could investigate this issue and provide a prompt solution.

**Thank you very much in advance!**

**Comment by AlexPardonXD** [comment]
Facing same issue on my android mobile, I am using metamask-sdk.js.  Its look like metmask app is not properly communicate with browser.  I opened dual window on my mobile. In one window brower was opened and In 2nd window metamask app was opened, then login was working fine.  I think issue arise when browser was running in background.

**Comment by codernuub** [comment]
Facing same issue on android mobile. I am using metamask-sdk.js.

Its look like metamask failing to communicate properly with dapp app, when browser running in background.

I opened my dapp webapp and metmask app on dual screen on same mobile, then login was working fine.

**Comment by charliepank** [comment]
Still exists on 7.56.5

**Comment by Kalovelo** [comment]
Same here on ios, any update on this?
